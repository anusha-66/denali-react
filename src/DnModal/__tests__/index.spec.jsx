/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnModal, DnButton, DnIcon } from '../../index'; // src/index.tsx

describe('DnModal Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnModal>
                <p>Dummy Component</p>
            </DnModal>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnModal
                id="modal1"
                isFull
                modalContainerClassName="w-600px"
                closeContent={
                    <DnButton>
                        <DnIcon icon="close" />
                    </DnButton>
                }
                headerContent={<span>Dummy Header</span>}
                footerContent={<span>Dummy Footer</span>}
            >
                <p>Dummy Component</p>
            </DnModal>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
