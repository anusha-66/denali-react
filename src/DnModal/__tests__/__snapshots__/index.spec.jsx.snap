// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnModal Tests should render correctly with custom props 1`] = `
<div
  class="modal is-active"
  id="modal1"
>
  <div
    class="modal-container is-full w-600px"
  >
    <div
      class="close"
    >
      <button
        class="button is-solid is-default"
        type="button"
      >
        <i
          class="d-icon d-close is-default"
        />
      </button>
    </div>
    <div
      class="modal-header"
    >
      <span>
        Dummy Header
      </span>
    </div>
    <div
      class="modal-content"
    >
      <p>
        Dummy Component
      </p>
    </div>
    <div
      class="modal-footer"
    >
      <span>
        Dummy Footer
      </span>
    </div>
  </div>
</div>
`;

exports[`DnModal Tests should render with default props 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-container"
  >
    <div
      class="modal-content"
    >
      <p>
        Dummy Component
      </p>
    </div>
  </div>
</div>
`;
